lib/steps.js

```diff
@@ -576,6 +577,108 @@ function detectCompendium(passon) {
     });
 }
 
+function fetchCompendiumID(passon) {
+    return new Promise((fulfill, reject) => {
+        if (passon.isCompendium) {
+            debug('[%s] Reading compendium ID from %s', passon.id, config.compendium.detectionFileName);
+
+            let compendiumID = null;
+
+            // Get compendium yml, or throw exception on error
+            try {
+                let doc = yaml.safeLoad(fs.readFileSync(passon.detectionFile, 'utf8'));
+                compendiumID = doc.id;
+                debug('[%s] Successfully read compendium metadata: %O', passon.id, doc);
+
+            } catch (error) {
+                debug('[%s] Could not read compendium detection file %s: %s', passon.id, passon.detectionFile, error);
+                error.msg = 'Could not read compendium detection file';
```

Comment:
You're not reading the compendium detection file, but the "compendium configuration file", see http://o2r.info/erc-spec/spec/#erc-configuration-file
 
Before it was only _used_ for detection. Please rephrase.